src/Inputs/scene.inputManager.ts

```diff
@@ -804,64 +806,81 @@ export class InputManager {
             }
         };
 
-        let attachedFunction = () => {
-            if (!elementToAttachTo || this._keyboardIsAttached) {
-                return;
-            }
-            elementToAttachTo.addEventListener("keydown", this._onKeyDown, false);
-            elementToAttachTo.addEventListener("keyup", this._onKeyUp, false);
-            this._keyboardIsAttached = true;
-        };
+        this._deviceSourceManager.onDeviceConnectedObservable.add((device) => {
+            // Keyboard Events
+            if (device.deviceType == DeviceType.Keyboard) {
+                this._deviceSourceManager.getDeviceSource(device.deviceType, device.deviceSlot)!.onInputChangedObservable.add((deviceSource) => {
+                    if (elementToAttachTo && document.activeElement === elementToAttachTo) {
+                        // Get Keyboard key and create event init data
+                        const key = String.fromCharCode(deviceSource.inputIndex);
+                        const init = {key: key, keyCode: deviceSource.inputIndex};
+
+                        if (deviceSource.currentState == 1) {
+                            const evt = new KeyboardEvent("keydown", init);

```

Comment:
why do we need to create a KeyboardEvent? Can't we reuse the source one?

Reply:
This will be changed to use something similar to PointerEvent because Babylon Native will not know what a KeyboardEvent is.